### ISSUE TITLE
Add additional channel to pre-trained resnet-50 model

### ISSUE THREAD

**Issue by jkini** [tensorflow/models#7707]
I am trying to use pre-trained Resnet50 model. However, I intend to modify the first convolution layer to be able to take a 4-channel tensor(image+binary mask), where the network weights are initialized from the pre-trained model, except for the newly added filters which are initialized randomly.

Adding this feature to existing Resnet models would be really helpful for segmentation tasks.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by jkini** [comment]
Updated the post to comply to the issue template.
